src/tests/cxxfrontend.c

```diff
@@ -253,23 +253,38 @@ void test_target()
 
 void test_emplace()
 {
-  Loc loc;
-  UnionExp ue;
-
-  IntegerExp::emplacei(&ue, loc, 1065353216, Type::tint32);
-  Expression *e = ue.exp();
-  assert(e->op == TOKint64);
-  assert(e->toInteger() == 1065353216);
-
-  UnionExp ure;
-  Expression *re = Compiler::paintAsType(&ure, e, Type::tfloat32);
-  assert(re->op == TOKfloat64);
-  assert(re->toReal() == CTFloat::one);
-
-  UnionExp uie;
-  Expression *ie = Compiler::paintAsType(&uie, re, Type::tint32);
-  assert(ie->op == TOKint64);
-  assert(ie->toInteger() == e->toInteger());
+    Loc loc;
+    UnionExp ue;
+
+    IntegerExp::emplacei(&ue, loc, 1065353216, Type::tint32);
+    Expression *e = ue.exp();
+    assert(e->op == TOKint64);
+    assert(e->toInteger() == 1065353216);
+
+    UnionExp ure;
+    Expression *re = Compiler::paintAsType(&ure, e, Type::tfloat32);
+    assert(re->op == TOKfloat64);
+    assert(re->toReal() == CTFloat::one);
+
+    UnionExp uie;
+    Expression *ie = Compiler::paintAsType(&uie, re, Type::tint32);
+    assert(ie->op == TOKint64);
+    assert(ie->toInteger() == e->toInteger());
+}
+
+/**********************************/
+
+void test_parameters()
+{
+    Parameters *args = new Parameters;
+    args->push(Parameter::create(STCundefined, Type::tint32, NULL, NULL, NULL));
+    args->push(Parameter::create(STCundefined, Type::tint64, NULL, NULL, NULL));
+
+    TypeFunction *tf = TypeFunction::create(args, Type::tvoid, VARARGnone, LINKc);
```

Comment:
This appears to fail on OSX

Reply:
In globals.d:
```
alias dinteger_t = ulong;
alias sinteger_t = long;
alias uinteger_t = ulong;
```
Mangling of these types is defined in cppmangle.d as:
```
case Tint64:
    c = target.c_longsize == 8 ? 'l' : 'x';
    break;
case Tuns64:
    c = target.c_longsize == 8 ? 'm' : 'y';
```

However in globals.h
```
#if __LP64__
typedef unsigned long dinteger_t;
typedef long sinteger_t;
typedef unsigned long uinteger_t;
#else
typedef unsigned long long dinteger_t;
typedef long long sinteger_t;
typedef unsigned long long uinteger_t;
#endif
```

Changing the condition in globals.h to match cppmangle.d
```
#if __SIZEOF_LONG__ == 8
...
```

Reply:
Hmm, but that doesn't work because @WalterBright changed that logic, so every compiler between 2.079.0 and 2.081.1 too broken to be used for building dmd.